src/main/java/frc/robot/subsystems/arm/Arm.java

```diff
@@ -1,9 +1,135 @@
+
+        initArmTalons();
+    }
+    
+    private void initArmTalons() {
+        TalonFXConfiguration toApply = new TalonFXConfiguration();
+
+        toApply.MotorOutput.Inverted = InvertedValue.Clockwise_Positive;
+        toApply.CurrentLimits.SupplyCurrentLimit = ArmCal.ARM_SUPPLY_CURRENT_LIMIT_AMPS;
+        toApply.CurrentLimits.StatorCurrentLimit = ArmCal.ARM_STATOR_CURRENT_LIMIT_AMPS;
+        toApply.CurrentLimits.StatorCurrentLimitEnable = true;
+        toApply.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+        toApply.Slot0.kP = ArmCal.ARM_MOTOR_P;
+        toApply.Slot0.kI = ArmCal.ARM_MOTOR_I;
+        toApply.Slot0.kD = ArmCal.ARM_MOTOR_D;
+        toApply.Slot0.kV = ArmCal.ARM_MOTOR_FF;
+
+        armMotorLeft.getConfigurator().apply(toApply);
+        armMotorRight.setControl(new Follower(armMotorLeft.getDeviceID(), true));
+    }
+
+    public void setDesiredPosition(ArmPosition armPosition) {
+        // If arm is within interference zone, don't set desired position
+        if (isArmMoveable()) {
+            this.armDesiredPosition = armPosition;
+        }
+    }
+
+    public boolean isArmMoveable() {
+        // return opposite of whether or not arm is in interference zone
+        return !isArmInInterferenceZone();
+    }
+
+    public void rezeroArm() {
+        armMotorLeft.setPosition(armLefttEncoderAbs.getAbsolutePosition().getValueAsDouble() * 360);
+    }
+
+    // Account for PID when setting position of our arm
+    public void controlPosition(double inputPositionDegrees) {
+        PositionVoltage desiredVoltage = new PositionVoltage(inputPositionDegrees / 360).withSlot(0);
+        armMotorRight.setControl(desiredVoltage);
+        // desired pos = requested pos
+        this.desiredSetpointPosition = inputPositionDegrees;
+        // convert velocity in rotations/sec to deg/sec
+        this.desiredSetpointVelocity = armMotorRight.getVelocity().getValueAsDouble() * 360;
+    }
+
+    public boolean atPositionDegrees(ArmPosition pos) {
+        double checkPositionDeg = armPositions.get(pos);
+        double currentPositionDeg = armMotorLeft.getPosition().getValueAsDouble() * 360;
+
+        return Math.abs(checkPositionDeg - currentPositionDeg) < ArmCal.ARM_MARGIN_DEGREES;
+    }
+    public boolean isArmInInterferenceZone() {
+        double currentPosition = armMotorLeft.getPosition().getValueAsDouble() * 360;
+        if (currentPosition <= ArmCal.ARM_INTERFERENCE_THRESHOLD_MAX_DEGREES && 
+            currentPosition >= ArmCal.ARM_INTERFERENCE_THRESHOLD_MIN_DEGREES 
+            /** TODO: AND is Deep Climb engaged */ ) {
+            return true;
+        }
+        return false;
+    }
     
-}
+
+    public void stopArmMovement() {
+        // left motor follows right motor, so armMotorLeft is not necessary here
+        armMotorLeft.setVoltage(0.0);
+    }
+
+    @Override
+    public void periodic() {
+        // control arm position
+        if (isArmMoveable()) {
```

Comment:
controlPosition(armPositions.get(this.armDesiredPosition));
^ just needs to be this line